plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -395,7 +398,7 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
-    public record CoercionOption(HiveTimestampPrecision timestampPrecision)
+    public record CoercionOption(HiveTimestampPrecision timestampPrecision, boolean treatNaNAsNull)
```

Comment:
Maybe `CoercionOption` should actually be `CoercionOptions` since it will contain multiple options?

Reply:
I have renamed it to `CoercionContext` WDYT ?